include/aspect/particle/output/hdf5.h

```diff
@@ -104,12 +102,13 @@ namespace aspect
           void
           load (std::istringstream &is);
 
-        private:
           /**
-           * Internal index of file output number.
+           * Returns a constructed filename.
            */
-          unsigned int file_index;
+          const std::string
+          get_file_name ();
```

Comment:
It makes no difference whether the string you return here is const or not. `const` only makes a difference for reference and pointer types when returned from a function. Can you remove the qualification?